ccmbioinfo/lfs
# CARDS-1998: When generating a new form from the backend, generate blank answers for all questions

With this, everything should be working fine in prems mode, either with manually created visits, or with imported visits.

To test in standalone mode:
- manually create  Patient Information and Visit Information forms
- check that these forms are correctly marked as complete and there are no duplicate answers
- check that the right data forms are auto-created
- check that the reference and computed answers have the right value (in the JSON or Composum, since they are hidden in the UI) 
- generate a token, access the patient portal with it, make sure the survey goes to the welcome screen
- fill in the survey as the patient, make sure it is correctly marked as incomplete when leaving unanswered questions, and correctly completed when all answers are entered
- make sure to test CPESIC too, since it has conditional sections

To test clarity import:
- build with `-Pdocker`
- run in `compose-cluster`: `python3 generate_compose_yaml.py --mssql --cards_project cards4prems --oak_filesystem --dev_docker_image --smtps --smtps_test_container --smtps_test_mail_path ~/cardsmail --composum --adminer` (you can use a different path than `~/cardsmail` if that one is not working correctly)
- run `python3 generate_test_sql.py -n 10 ./importTestSql.new.sql` in `compose-cluster/mssql`
- start docker with `docker-compose up`
- at http://localhost:1435/ import the generated sample data file from `compose-cluster/mssql/importTestSql.new.sql`
- at http://localhost:8080/system/console/configMgr change the default frequency for `Clarity import filter - CPESIC percentage` to `1.0`
- open http://localhost:8080/Subjects.importClarity
- check as above that the forms are imported correctly, and at least one of them can be filled in and submitted by a patient with a token

Question 1: the following behavior is new: when creating a new form with no mandatory questions (like Visit Information), the form is no longer created as a Draft/Incomplete, since all 0 mandatory questions have been answered. Should this revert to the old behavior, new forms are always incomplete? If we force that new forms are incomplete, the CSV import will start creating only draft forms.

Question 2: the following behavior might also be changed: as I remember, CSV imported forms were never marked as incomplete, since the agreement was that data imported is considered complete, even if mandatory questions are not answered, but the new code always computes flags correctly; do we want to revert to the old convention that imported data is considered complete?
